[python] Scientific Computing with Python projects test output

### Describe the Issue

The new platform for the Scientific Computing with Python projects does not give very much feedback. Without the diff and assertion output from the tests on replit it's extremely difficult to know what the problem is.

**Case 1**
For example this error:
>"create_spend_chart should print a different chart representation. Check that all spacing is exact."

https://forum.freecodecamp.org/t/scientific-computing-with-python-projects-budget-app/677107/5

There is no other information at all in the preview or console areas. It's not even clear if the calculation is correct or if it's just a spacing issue. 

To solve this I had to look for the test data and expected output on github to find that the problem was an extra newline at the very end (pretty common for this project.) I also use `repr` to print raw string data to compare, so you can see newlines clearly, but that's never introduced.

**Case 2** 
Here is another example:
https://forum.freecodecamp.org/t/scientific-computing-with-python-projects-arithmetic-formatter/676946

Again, without the diff and assertion output these are extremely more difficult. It was already difficult for learners to understand that error output without a primer.

**Case 3**
In this case there was a syntax error that prevented any code from running, but no output at all to determine the problem.
https://forum.freecodecamp.org/t/scientific-computing-with-python-projects-budget-app/676810

A "There was a syntax error before any code could be run" message or a real Python syntax error would be helpful here.
```
line 9
    amount_str = f' {format(entry["amount"], '.2f')}'
                                               ^
SyntaxError: invalid decimal literal
```

Proposed solution:

If the fails/hints could output actual and expected raw string data in the console, that would be something to go by.

Additionally, it would help if the example in the instructions had explicit information about how many spaces there need to be, if it should end with a space, or newline, etc.

### Affected Page

https://www.freecodecamp.org/learn/scientific-computing-with-python/scientific-computing-with-python-projects/budget-app

### Your code

```

NA

```


### Expected behavior

NA


### Screenshots

NA


### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]
NA


### Additional context

NA

The test is too sensitive to trailing blank spaces:
pyodide.asm.js:9 AssertionError: '\nPercentage spent by category\n100|\n 90[298 chars]    ' != 'Percentage spent by category\n100|       [383 chars] t  '
Even looking at the console as a student I don't know what that means!

TODO: test should ignore spaces at the end of each line

--- a/curriculum/challenges/english/07-scientific-computing-with-python/build-a-budget-app-project/budget-app.md
+++ b/curriculum/challenges/english/07-scientific-computing-with-python/build-a-budget-app-project/budget-app.md
@@ -77,6 +77,8 @@ Percentage spent by category
         g     
 ```
 
+Note: open the browser console with F12 to see a more verbose output of the tests.
+
 # --hints--
 
 The `deposit` method should create a specific object in the ledger instance variable.

--- a/curriculum/challenges/english/07-scientific-computing-with-python/build-a-polygon-area-calculator-project/polygon-area-calculator.md
+++ b/curriculum/challenges/english/07-scientific-computing-with-python/build-a-polygon-area-calculator-project/polygon-area-calculator.md
@@ -75,6 +75,8 @@ Square(side=4)
 8
 ```
 
+Note: open the browser console with F12 to see a more verbose output of the tests.
+
 # --hints--
 
 The `Square` class should be a subclass of the `Rectangle` class.

--- a/curriculum/challenges/english/07-scientific-computing-with-python/build-a-probability-calculator-project/probability-calculator.md
+++ b/curriculum/challenges/english/07-scientific-computing-with-python/build-a-probability-calculator-project/probability-calculator.md
@@ -55,6 +55,7 @@ Since this is based on random draws, the probability will be slightly different 
 
 _Hint: Consider using the modules that are already imported at the top. Do not initialize random seed within the file._
 
+Note: open the browser console with F12 to see a more verbose output of the tests.
 
 # --hints--
 

--- a/curriculum/challenges/english/07-scientific-computing-with-python/build-a-time-calculator-project/time-calculator.md
+++ b/curriculum/challenges/english/07-scientific-computing-with-python/build-a-time-calculator-project/time-calculator.md
@@ -44,6 +44,8 @@ add_time('6:30 PM', '205:12')
 
 Do not import any Python libraries. Assume that the start times are valid times. The minutes in the duration time will be a whole number less than 60, but the hour can be any whole number.
 
+Note: open the browser console with F12 to see a more verbose output of the tests.
+
 # --hints--
 Calling `add_time('3:30 PM', '2:12')` should return `'5:42 PM'`.
 

--- a/curriculum/challenges/english/07-scientific-computing-with-python/build-an-arithmetic-formatter-project/arithmetic-formatter.md
+++ b/curriculum/challenges/english/07-scientific-computing-with-python/build-an-arithmetic-formatter-project/arithmetic-formatter.md
@@ -68,6 +68,8 @@ The function will return the correct conversion if the supplied problems are pro
   - There should be four spaces between each problem.
   - There should be dashes at the bottom of each problem. The dashes should run along the entire length of each problem individually. (The example above shows what this should look like.)
 
+Note: open the browser console with F12 to see a more verbose output of the tests.
+
 # --hints--
 
 `arithmetic_arranger(["3801 - 2", "123 + 49"])` should return `  3801      123\n-    2    +  49\n------    -----`.

